docs/standard/native-interop/tutorial-custom-marshaller.md

```diff
@@ -0,0 +1,278 @@
+```
+
+Next, consider the case where multiple instances of `error_data` are returned. Typically, you use a collection to return multiple elements. Since this marshaller will be returning multiple elements, this is called the `MarshalMode.ElementOut` mode.
+
+```csharp
+namespace CustomMarshalling
+{
+    [CustomMarshaller(typeof(ErrorData), MarshalMode.ManagedToUnmanagedIn, typeof(ErrorDataMarshaller))]
+    [CustomMarshaller(typeof(ErrorData), MarshalMode.ElementOut, typeof(Out))]
+    internal static unsafe class ErrorDataMarshaller
+    {
+        //
+        // Other marshallers removed
+        //
+
+        public static class Out
+        {
+            public static ErrorData ConvertToManaged(ErrorDataUnmanaged unmanaged)
+                => throw new NotImplementedException();
+
+            public static void Free(ErrorDataUnmanaged unmanaged)
+                => throw new NotImplementedException();
+        }
+    }
+}
+```
+
+The conversion from `ErrorDataUnmanaged` to `ErrorData` is the inverse of what you did for the "in" mode in the preceding code. Remember that you also need to "clean up" any allocations that the unmanaged environment expected you to perform.
+
+For the managed to unmanaged "out" marshaller, you're going to do something special. The name of the data type you're marshalling is called `error_data` and .NET typically expresses errors as exceptions. However, some errors are more impactful than others. Errors identified as "fatal" usually indicate a catastrophic or unrecoverable error. Notice the `error_data` has a field to check if the error is fatal. You'll marshal an `error_data` into managed code, and if it's fatal, you'll throw an exception rather than just converting it into an `ErrorData` and returning it.
+
+```csharp
+namespace CustomMarshalling
+{
+    [CustomMarshaller(typeof(ErrorData), MarshalMode.ManagedToUnmanagedIn, typeof(ErrorDataMarshaller))]
+    [CustomMarshaller(typeof(ErrorData), MarshalMode.ElementOut, typeof(Out))]
+    [CustomMarshaller(typeof(ErrorData), MarshalMode.ManagedToUnmanagedOut, typeof(ThrowOnFatalErrorOut))]
+    internal static unsafe class ErrorDataMarshaller
+    {
+        //
+        // Other marshallers removed
+        //
+
+        public static class ThrowOnFatalErrorOut
+        {
+            public static ErrorData ConvertToManaged(ErrorDataUnmanaged unmanaged)
+                => throw new NotImplementedException();
+
+            public static void Free(ErrorDataUnmanaged unmanaged)
+                => throw new NotImplementedException();
+        }
+    }
+}
+```
+
+An "out" parameter converts from an unmanaged context into a managed context, so you implement the `ConvertToManaged` method. When the unmanaged callee returns and provides an `ErrorDataUnmanaged` object, you can inspect it using your `ElementOut` mode marshaller and then check if it's marked as a "fatal" error. If so, that's your indication to throw as opposed to just returning the `ErrorData`.
```

Comment:
```suggestion
An "out" parameter converts from an unmanaged context into a managed context, so you implement the `ConvertToManaged` method. When the unmanaged callee returns and provides an `ErrorDataUnmanaged` object, you can inspect it using your `ElementOut` mode marshaller and check if it's marked as a fatal error. If so, that's your indication to throw instead of just returning the `ErrorData`.
```